lib/common/http1client.c

```diff
@@ -454,18 +455,32 @@ static void on_head(h2o_socket_t *sock, const char *err)
             headers[i].flags = (h2o_header_flags_t){0};
         }
 
-        if (!(100 <= http_status && http_status <= 199 && http_status != 101))
+        if (http_status == 101) {
+            assert(!client->_use_expect);
```

Comment:
@kazuho One question: this means that a malicious upstream server can intentionally crash h2o by sending a 101 response to h2o that sent an 'expect' header. Is this safe?

Reply:
Of course you are correct. Fixed in 5f695e4 and 29f807e.